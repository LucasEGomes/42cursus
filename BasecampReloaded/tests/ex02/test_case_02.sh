#!/bin/sh
mkdir -p temp
mkdir -p temp/folder
mkdir -p temp/folder\~
mkdir -p temp/\#folder\#
touch temp/folder/\#delete_me{0..5}\#
touch temp/folder/delete_me{0..5}\~
touch temp/folder/do_not_delete_me{0..5}\#
touch temp/folder\~/do_not_delete_me{0..5}
touch temp/\#folder\#/do_not_delete_me{0..5}
cp $ROOT_DIR/ex02/clean temp/
cd temp
bash clean | sort > ../test_case_02.user
cd ..
diff test_case_02.expected test_case_02.user > test_case_02.output
RESULT=$(find temp | wc -l)
if [ -z "$(cat test_case_02.output)" ] && [ "$RESULT" -eq 23 ];
then
    echo -n "O" >> $TEST_SUMMARY
else
    echo -n "X" >> $TEST_SUMMARY
fi
rm -rf temp/
