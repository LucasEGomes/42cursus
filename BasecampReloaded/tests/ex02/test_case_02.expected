./folder/#delete_me0#
./folder/#delete_me1#
./folder/#delete_me2#
./folder/#delete_me3#
./folder/#delete_me4#
./folder/#delete_me5#
./folder/delete_me0~
./folder/delete_me1~
./folder/delete_me2~
./folder/delete_me3~
./folder/delete_me4~
./folder/delete_me5~
